tool/grind.dart

```diff
@@ -59,8 +61,64 @@ build(GrinderContext context) {
 }
 
 /// Prepare the app for deployment.
-void deploy(GrinderContext context) {
-  context.log('execute: `appcfg.py update build/web`');
+deploy(GrinderContext context) {
+  // Validate the deloy. This means that we're using version `dev` on the master
+  // branch and version `prod` on the prod branch. We only deploy prod from the
+  // prod branch. Other versions are possible but not verified.
+
+  // `dev` is served from dev.dart-pad.appspot.com
+  // `prod` is served from prod.dart-pad.appspot.com and from dartpad.dartlang.org.
+
+  Map app = yaml.loadYaml(new File('web/app.yaml').readAsStringSync());
+
+  final String version = app['version'];
+  List handlers = app['handlers'];
+  bool isSecure = false;
+
+  for (Map m in handlers) {
+    if (m['url'] == '.*') {
+      isSecure = m['secure'] == 'always';
+    }
+  }
+
+  return GitDir.fromExisting('.').then((GitDir dir) {
+    return dir.getCurrentBranch();
+  }).then((BranchReference branchRef) {
+    final String branch = branchRef.branchName;
+
+    context.log('branch: ${branch}');
+    context.log('version: ${version}');
+
+    if (branch == 'prod') {
+      if (version != 'prod') {
+        context.fail('Trying to deploy non-prod version from the prod branch');
```

Comment:
Consider making this a warning not a fail if there's an easy way to do. We may want to do this to do testing immediately before release